### ISSUE TITLE
was_ami search issue for windows based AMI when using latest(1.21) EKS version

### ISSUE THREAD

**Issue by vmasule** [terraform-aws-modules/terraform-aws-eks#1510]
## Description

With previous version it is working fine but now when I tried creating EKS cluster with version 1.21 I am getting some AMI search error and that's blocking me from running my existing scripts, I have even passed specific AMI id for windows Worker Groups but still the same error.

Before you submit an issue, please perform the following first:

1. Remove the local `.terraform` directory (! ONLY if state is stored remotely, which hopefully you are following that best practice!): `rm -rf .terraform/`
2. Re-initialize the project root to pull down modules: `terraform init`
3. Re-attempt your terraform plan or apply and check if the issue still persists

## Versions
```
terraform {
  required_version = ">= 0.13.1"

  required_providers {
    aws        = ">= 3.40.0"
    local      = ">= 1.4"
    kubernetes = {
      source  = "hashicorp/kubernetes"
      version = ">= 2.0"
    }
    http = {
      source  = "terraform-aws-modules/http"
      version = ">= 2.4.1"
    }
  }
}
```

## Reproduction

```
  worker_groups_launch_template = [
    {
      name                          = "windows-ng"
      ami_id                        = "ami-07e24fe860ec36f19"
      platform                      = "windows"
      instance_type                 = "t2.medium"
      asg_desired_capacity          = 2
      asg_max_size                  = 3
      asg_min_size                  = 2
      subnets                       = var.windows_subnets
    }
  ]
```

## Expected behavior
Should allow to create EKS cluster

## Actual behavior
Getting below error
```
╷
│ Error: Your query returned no results. Please change your search criteria and try again.
│ 
│   with module.eks.data.aws_ami.eks_worker_windows[0],
│   on .terraform/modules/eks/data.tf line 33, in data "aws_ami" "eks_worker_windows":
│   33: data "aws_ami" "eks_worker_windows" {

```

## Additional context
It seems for 1.21 this module try to search for default windows AMI but those are not updated by AWS yet and hence resulting in this error, the first filter is the issue here.
```
data "aws_ami" "eks_worker_windows" {
  count = local.worker_has_windows_ami ? 1 : 0

  filter {
    name   = "name"
    values = [local.worker_ami_name_filter_windows]
  }

  filter {
    name   = "platform"
    values = ["windows"]
  }

  most_recent = true

  owners = [var.worker_ami_owner_id_windows]
}
```

**Comment by StefanCzubek** [comment]
Internally, the EKS-Module tries to find an AMI that matches the cluster-version:

(local.tf)
`  worker_ami_name_filter_windows = (var.worker_ami_name_filter_windows != "" ?
    var.worker_ami_name_filter_windows : "Windows_Server-2019-English-Core-EKS_Optimized-${tonumber(var.cluster_version) >= 1.14 ? var.cluster_version : 1.14}-*"
  )`

It seems like AWS has not yet released a matching AMI for K8S version 1.21 (which is a bit strange, since EKS Version 1.21 was released about 3 weeks ago):
![image](https://user-images.githubusercontent.com/49159002/128341731-0b1bf9df-7fbd-4c2a-8b4b-1d9796ab5545.png)


As a "quickfix" you can override the variable "worker_ami_name_filter_windows" to use version 1.20 instead. However, you should delete this quickfix when AWS has released a matching AMI for 1.21 (see AWS->EC2-Dashboard->AMIs-> Search for Prefix Windows_Server-2019-English-Core-EKS_Optimized-)

Currently working snippet for using the module with cluster_version=1.21 (shortened):

> module "eks" {
>   source  = "terraform-aws-modules/eks/aws"
> 
>   cluster_version                              = "1.21"
> 
>   worker_ami_name_filter_windows               = "Windows_Server-2019-English-Core-EKS_Optimized-1.20-*" /
> }
>

**Comment by replicant0wnz** [comment]
So we don't use *any* Windows worker nodes and we still get this error. Doing the:

`worker_ami_name_filter_windows = "Windows_Server-2019-English-Core-EKS_Optimized-1.20-*" /`

Does fix it though :-|

**Comment by StefanCzubek** [comment]
Update: Seems like there is now an AMI for EKS 1.21 available:

![image](https://user-images.githubusercontent.com/49159002/131331322-30bc727f-822e-484d-ad2d-458b1f0f1a55.png)

**Comment by daroga0002** [comment]
Thank you @StefanCzubek for investigation. So this seems be a issue on AWS side, probably Windows/EKS/Linux teams are not syncing each other on releases 👎 

You can always open a Issue to them under https://github.com/aws/containers-roadmap/issues as such stories were already happening in other areas.

I am closing this issue

**Comment by vmasule** [comment]
Yeah , I can confirm this been fixed as I was able to migrate to K8s 1.21 version.

**Comment by darkn3rd** [comment]
I am having this problem with Linux.
